src/broker.rs

```diff
@@ -0,0 +1,77 @@
+use crate::MQTT_INSECURE_DEFAULT_PORT;
+use embedded_nal::{nb, AddrType, Dns, IpAddr, Ipv4Addr, SocketAddr};
+
+pub trait Broker {
+    fn get_address(&mut self) -> Option<SocketAddr>;
+    fn set_port(&mut self, port: u16);
+}
+
+pub struct NamedBroker<R: Dns> {
+    raw: &'static str,
```

Comment:
Yeah this is just me being lazy as I initially test things. We can give it the `'buf` lifetime without issue I think.

Reply:
Hmm, after playing around with this, I don't think it's possible to use a lifetime specifier that's unknown. It has to be `'static`, or we have to maintain a copy of the value.

This happens because the lifetime propogates all the way out to `Shared` resource structures in RTIC, where the lifetime has to be known. Perhaps this indicates that the `broker` name should be stored as a `String` instead, perhaps with a user-specified size.